paimon-format/src/main/java/org/apache/paimon/format/parquet/reader/ParquetColumnVector.java

```diff
@@ -67,7 +75,20 @@ public class ParquetColumnVector {
             return;
         }
 
-        if (isPrimitive) {
+        if (column.variantFileType().isPresent()) {
+            ParquetField fileContentCol = column.variantFileType().get();
+            WritableColumnVector fileContent =
+                    ParquetSplitReaderUtil.createWritableColumnVector(
```

Comment:
Here I need a clean `createWritableColumnVector` method. like Spark's: `public OffHeapColumnVector(int capacity, DataType type)` 

But this method in paimon seems to require parquetType, therefore, I have no choice but to introduce the `parquetType` type in `ParquetField`.

Reply:
Addressed, let's follow Spark